sprint_misison_7/queries.sql

```diff
@@ -0,0 +1,153 @@
+/*
+  다음 경우들에 대해 총 14개의 SQL 쿼리를 작성해 주세요.
+  예시로 값이 필요한 경우 적당한 값으로 채워넣어서 작성하면 됩니다. 
+*/
+
+/*
+  1. 내 정보 업데이트 하기
+  - 닉네임을 "test"로 업데이트
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+UPDATE users
+SET nickname = 'test'
+WHERE id = 1;
+
+
+/*
+  2. 내가 생성한 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/ 
+SELECT *
+FROM products
+WHERE seller_id = 1
+ORDER BY created_at DESC
+LIMIT 10 OFFSET 20;
+
+
+/*
+  3. 내가 생성한 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+SELECT COUNT(*) AS total_count
+FROM products
+WHERE seller_id = 1;
+
+
+/*
+  4. 내가 좋아요 누른 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+SELECT p.*
+FROM products p
+JOIN product_likes pl ON p.id = pl.product_id
+WHERE pl.user_id = 1
+ORDER BY pl.created_at DESC
+LIMIT 10 OFFSET 20;
+-- PK(user_id, product_id) 를 사용하여 좋아요 중복을 방지
+
+/*
+  5. 내가 좋아요 누른 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+SELECT COUNT(*) AS total_count
+FROM product_likes
+WHERE user_id = 1;
+
+
+/*
+  6. 상품 생성
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+INSERT INTO products (seller_id, category_id, name, description, price, condition, status)
+VALUES (1, 1, 'iPhone 15 Pro', '거의 새것이나 다름없는 상태입니다', 1200000, 'LIKE_NEW', 'FOR_SALE');
+RETURNING id;
+-- categories.id = 1 이 존재한다고 가정했을때
+-- RETURNING id 를 통해 생성된 상품의 id를 반환
+
+/*
+  7. 상품 목록 조회
+  - 상품명에 "test"가 포함된 상품 검색
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 1번째 페이지
+  - 각 상품의 좋아요 개수를 포함해서 조회하기
+*/
+SELECT p.*, p.like_count
+FROM products p
+WHERE LOWER(p.name) LIKE '%' || LOWER('test') || '%'
+ORDER BY p.created_at DESC
+LIMIT 10 OFFSET 0;
```

Comment:
대소문자 무시 검색이 목적이라면 `ILIKE`가 간결하고 의미도 분명할 것 같아요.

```suggestion
SELECT
  p.id, p.name, p.price, p.created_at, p.like_count
FROM products p
WHERE p.name ILIKE '%test%'
ORDER BY p.created_at DESC, p.id DESC
LIMIT 10 OFFSET 0;
```

Reply:
넵 확인했습니다